packages/interactivity/src/init.ts

```diff
@@ -29,6 +30,10 @@ export const initialVdom = new WeakMap< Element, ComponentChild[] >();
 
 // Initialize the router with the initial DOM.
 export const init = async () => {
+	// Parse and populate the initial state and config.
+	const data = parseInitialData();
+	populateInitialData( data );
+
```

Comment:
I had to move the state initialization here to prevent issues with circular references between modules. ☝️

Reply:
It was a real issue. IIRC, the `proxies/state.ts` module was `undefined` while trying to import `proxifyState` inside `store.ts`. The circular dependency was like this:

```
proxies/state > hooks > store > proxies/state
```

I fixed it  in https://github.com/WordPress/gutenberg/pull/62734/commits/e1694a6da4c1b43abe0cbe9f48ea26428b5068a1 by moving the scope and namespace-related logic to their respective modules.